src/components/SpaceProposalVote.vue

```diff
@@ -20,46 +20,52 @@ const selectedChoices = computed(() => {
 });
 
 const validatedUserChoice = computed(() => {
-  if (!props.userVote?.choice) return null;
+  if (!userVote.value?.choice) return null;
   if (
     voting[props.proposal.type].isValidChoice(
-      props.userVote.choice,
+      userVote.value.choice,
       props.proposal.choices
     )
   ) {
-    return props.userVote.choice;
+    return userVote.value.choice;
   }
   return null;
 });
 
 function emitChoice(c) {
   emit('update:modelValue', c);
 }
+
+watch(web3Account, loadUserVote, { immediate: true });
 </script>
 
 <template>
   <BaseBlock class="mb-4" :title="$t('proposal.castVote')">
     <div class="mb-3">
       <SpaceProposalVoteSingleChoice
         v-if="proposal.type === 'single-choice' || proposal.type === 'basic'"
+        :key="validatedUserChoice"
```

Comment:
What's the purpose of this `key`, when it's not in a loop ?

`validatedUserChoice` is potentially `null`, and `key` does not accept null type.

Reply:
The purpose is to update the component when when `validatedUserChoice` changes. e.g when block is mounted and also when user changes or connects a wallet.

Reply:
Typescript is complaining about its type

![Screenshot 2023-05-30 at 4 20 52 PM](https://github.com/snapshot-labs/snapshot/assets/495709/3361d758-25fc-4b8b-9cd9-78dc53287e8b)

Reply:
Fixes please review PR again